src/EventStore.Core.Tests/ClientAPI/ExpectedVersion64Bit/read_stream_with_link_to_event_with_event_number_greater_than_int_maxvalue.cs

```diff
@@ -0,0 +1,74 @@
+using EventStore.ClientAPI;
+using EventStore.Core.Data;
+using EventStore.Core.Services;
+using NUnit.Framework;
+using System.Linq;
+
+namespace EventStore.Core.Tests.ClientAPI.ExpectedVersion64Bit
+{
+    [TestFixture]
+    [Category("ClientAPI"), Category("LongRunning")]
+    public class read_stream_with_link_to_event_with_event_number_greater_than_int_maxvalue : MiniNodeWithExistingRecords
+    {
+        private const string StreamName = "read_stream_with_link_to_event_with_event_number_greater_than_int_maxvalue";
+        private const long intMaxValue = (long)int.MaxValue;
+
+        private string _linkedStreamName = "linked-" + StreamName;
+        private EventRecord _event1, _event2;
+
+        public override void WriteTestScenario()
+        {
+            _event1 = WriteSingleEvent(StreamName, intMaxValue + 1, new string('.', 3000));
+            _event2 = WriteSingleEvent(StreamName, intMaxValue + 2, new string('.', 3000));
+
+            WriteSingleEvent(_linkedStreamName, 0, string.Format("{0}@{1}", intMaxValue + 1, StreamName), eventType: SystemEventTypes.LinkTo);
+            WriteSingleEvent(_linkedStreamName, 1, string.Format("{0}@{1}", intMaxValue + 2, StreamName), eventType: SystemEventTypes.LinkTo);
+        }
+
+        public override void Given()
+        {
+            _store = BuildConnection(Node);
+            _store.ConnectAsync().Wait();
+        }
+
+        [Test]
+        public void should_be_able_to_read_link_stream_forward_and_resolve_link_tos()
+        {
+            var readResult = _store.ReadStreamEventsForwardAsync(_linkedStreamName, 0, 100, true, DefaultData.AdminCredentials).Result;
+            Assert.AreEqual(SliceReadStatus.Success, readResult.Status);
+            Assert.AreEqual(2, readResult.Events.Length);
+            Assert.AreEqual(_event1.EventId, readResult.Events[0].Event.EventId);
```

Comment:
The test `subscribe_to_stream_with_link_to_event_with_event_number_greater_than_int_maxvalue` checks the EventNumber, however these ones don't. Is there any reason for that?